src/System.Runtime.InteropServices/tests/System/Runtime/InteropServices/ICustomMarshalerTests.cs

```diff
@@ -0,0 +1,537 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+namespace System.Runtime.InteropServices.Tests
+{
+    public class ICustomMarshalerTests
+    {
+        // To avoid having to create a native test library to reference in tests that
```

Comment:
I think it'd be useful to create a native test library like this - it would allow us to test more stuff like marshalling structs and classes and arrays. However, I actually don't know the best way to create an XPlat runtime library and don't want to waste people's time as we focus on 2.0 things